packages/core/src/external/aws/lambda-logic/bundle-to-html-derm.ts

```diff
@@ -0,0 +1,2688 @@
+
+    return {
+      ...acc,
+      [id]: resource,
+    };
+  }, {});
+}
+
+// find condition with code Z00 in the past year
+function getDermConditions(conditions: Condition[]): Condition[] {
+  const dermConditions = conditions.filter(condition => {
+    const code = getSpecificCode(condition.code?.coding ?? [], [ICD_10_CODE]);
+
+    // Check if the code is in the L00-L99 range
+    const isDermCode = code ? /L\d{2}(?:\.\d+)?/i.test(code) : false;
+
+    return isDermCode;
+  });
+
+  return dermConditions;
+}
+
+function getRheumatoidConditions(conditions: Condition[]) {
+  const rheumatoidConditions = conditions.filter(condition => {
+    const code = getSpecificCode(condition.code?.coding ?? [], [ICD_10_CODE]);
+
+    // Check if the code is in the M00-M99 range
+    const isRheumatoidCode = code ? /M\d{2}(?:\.\d+)?/i.test(code) : false;
+
+    return isRheumatoidCode;
+  });
+
+  return rheumatoidConditions;
+}
+
+function getAsthmaConditions(conditions: Condition[]) {
+  const rheumatoidConditions = conditions.filter(condition => {
+    const code = getSpecificCode(condition.code?.coding ?? [], [ICD_10_CODE]);
+
+    // Check if the code is in the L00-L99 range
+    const isAsthmaCode = code ? code.includes("J45") : false;
+
+    return isAsthmaCode;
+  });
+
+  return rheumatoidConditions;
+}
+
+function hasClinicalRelevantData(fhirTypes: FhirTypes): boolean {
+  const hasValues: string[] = [];
+
+  Object.entries(fhirTypes).forEach(([key, value]) => {
+    const isNotRelatedPersons = key !== "relatedPersons";
+    const isNotCoverages = key !== "coverages";
+    const hasValue = value && Array.isArray(value) && value.length;
+
+    if (isNotRelatedPersons && isNotCoverages && hasValue) {
+      hasValues.push(key);
+    }
+  });
+
+  return hasValues.length > 0;
+}
+
+function getValidCode(coding: Coding[] | undefined): Coding[] {
+  if (!coding) return [];
+
+  return coding.filter(coding => {
+    return (
+      coding.code &&
+      coding.code !== UNK_CODE &&
+      coding.display &&
+      coding.display.toLowerCase() !== UNKNOWN_DISPLAY
+    );
+  });
+}
+
+type ChartData = {
+  labels: string[];
+  data: number[];
+  min?: number;
+  max?: number;
+};
+
+type ObsSummary = {
+  effectiveDate: string;
+  vitalsValue: string;
+};
+
+function creteBPChartSection(observations: Observation[]) {
+  if (!observations) {
+    return {
+      section: "",
+      chartSystolicData: { labels: [], data: [] },
+      chartDiastolicData: { labels: [], data: [] },
+    };
+  }
+
+  const systolicObservations = observations
+    .filter(observation => {
+      const observationDisplay = observation.code?.coding?.find(coding => {
+        const code = fetchCodingCodeOrDisplayOrSystem(coding, "code");
+        return code === "8480-6";
+      });
+
+      return !!observationDisplay;
+    })
+    .filter(observation => {
+      return dayjs(observation.effectiveDateTime).isAfter(dayjs().subtract(2, "year"));
+    })
```

Comment:
nit: since we have the whole code here (not separated functions), we could just include the second filter on the first one, avoiding additional iterations.

here and below